src/basetypes/BondGraph.jl

```diff
@@ -62,15 +63,12 @@ struct BondGraphNode <: AbstractNode
     bondgraph::BondGraph
     type::Symbol
     name::Symbol
+    exposed::Vector{SourceSensor}
     freeports::Vector{Bool}
     vertex::RefValue{Int}
-    parameters::Vector{Num}
-    states::Vector{Num}
-    equations::Vector{Equation}
-    function BondGraphNode(bg::BondGraph, type = :BG, name = bg.name; vertex::Int = 0)
-        new(bg, Symbol(type), Symbol(name), Bool[], Ref(vertex),
-            parameters(bg), states(bg), equations(bg))
-    end
+end
+function BondGraphNode(bg::BondGraph, type = :BG, name = bg.name; vertex::Int = 0)
```

Comment:
add bool for creating a copy of bond graph (instead of linking to the same copy)

Reply:
deepcopy